### ISSUE TITLE
kiloview n60 not login

### ISSUE THREAD

**Issue by ctomast** [bitfocus/companion-module-kiloview-ndi#10]
i can't login in a kiloview n60. 

24.11.27 21:11:53 Instance/Wrapper/kiloview: Could not reach device. Retrying in 30 seconds.

Current firmware version： 2.02.0024
Current hardware version：1.0

In the login window i need to check " User License Agreement" if not, don't login

**Comment by josephdadams** [comment]
Are you saying that it works fine if you check the "User License Agreement"?

**Comment by ctomast** [comment]
Hi, When you access the device via web, you have to mark the check to be able to access, when I access via bitfocus, it always says that it is not accessible.   " could this be the error??"

24.12.10 08:07:26 Instance/Wrapper/kiloview: Could not reach device. Retrying in 30 seconds.

Thanks

De: Joseph Adams ***@***.***>
Enviat: dimarts, 10 de desembre de 2024 0:46
Per a: bitfocus/companion-module-kiloview-ndi ***@***.***>
A/c: Tomas Comellas, Carles ***@***.***>; Author ***@***.***>
Tema: Re: [bitfocus/companion-module-kiloview-ndi] kiloview n60 not login (Issue #10)


Are you saying that it works fine if you check the "User License Agreement"?

-

**Comment by josephdadams** [comment]
I don’t think so. In my experience, that had to be done every time anyway. Did you create an API user in the Kiloview web interface and are you using those credentials in the module?

**Comment by ctomast** [comment]
Hello, I can create users on the kiloview web interface , but it still does not allow me access via companion.

What do you mean by an API user?  Where is it created?

Thank you

De: Joseph Adams ***@***.***>
Enviat: dimarts, 10 de desembre de 2024 12:57
Per a: bitfocus/companion-module-kiloview-ndi ***@***.***>
A/c: Tomas Comellas, Carles ***@***.***>; Author ***@***.***>
Tema: Re: [bitfocus/companion-module-kiloview-ndi] kiloview n60 not login (Issue #10)


I don't think so. In my experience, that had to be done every time anyway. Did you create an API user in the Kiloview web interface and are you using those credentials in the module?

-

**Comment by josephdadams** [comment]
Same place you create users. I believe it is a setting there.

**Comment by ctomast** [comment]
Hello, I don't see any configuration to create an API user.

***@***.***

De: Joseph Adams ***@***.***>
Enviat: dimarts, 10 de desembre de 2024 15:47
Per a: bitfocus/companion-module-kiloview-ndi ***@***.***>
A/c: Tomas Comellas, Carles ***@***.***>; Author ***@***.***>
Tema: Re: [bitfocus/companion-module-kiloview-ndi] kiloview n60 not login (Issue #10)


Same place you create users. I believe it is a setting there.

-

**Comment by DJEX93** [comment]
> Hello, I don't see any configuration to create an API user.
> 
> ***@***.***
> 
> De: Joseph Adams ***@***.***>
> Enviat: dimarts, 10 de desembre de 2024 15:47
> Per a: bitfocus/companion-module-kiloview-ndi ***@***.***>
> A/c: Tomas Comellas, Carles ***@***.***>; Author ***@***.***>
> Tema: Re: [bitfocus/companion-module-kiloview-ndi] kiloview n60 not login (Issue [#10](https://github.com/bitfocus/companion-module-kiloview-ndi/issues/10))
> 
> 
> Same place you create users. I believe it is a setting there.
> 
> -
> Reply to this email directly, view it on GitHub<https://cas5-0-urlprotect.trendmicro.com:443/wis/clicktime/v1/query?url=https%3a%2f%2fgithub.com%2fbitfocus%2fcompanion%2dmodule%2dkiloview%2dndi%2fissues%2f10%23issuecomment%2d2531849732&umid=7847b9ea-f140-4006-bab7-4686ff1f6c0e&auth=84fd7b44f3c6e5843978f04d97647fc72161b300-182b87fb8eacad2d41e15435fc067b6d0cea588d>, or unsubscribe<https://cas5-0-urlprotect.trendmicro.com:443/wis/clicktime/v1/query?url=https%3a%2f%2fgithub.com%2fnotifications%2funsubscribe%2dauth%2fBLFYR4HQOSC4SNBEAF5CCAL2E35GLAVCNFSM6AAAAABSTUUKS6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMZRHA2DSNZTGI&umid=7847b9ea-f140-4006-bab7-4686ff1f6c0e&auth=84fd7b44f3c6e5843978f04d97647fc72161b300-7ca50389a71d766623c6f2bd2a2187920694b539>.
> You are receiving this because you authored the thread.Message ID: ***@***.***>

I'agree with you nothing on this user page

**Comment by McHauge** [comment]
Tested with module v2.1.1

Can confirm it's working for me with:
 - N4: v2.02.0227 (API v1.0)

Can confirm that the module does not work for me with:
 - N50: v2.01.0007 (API v2.1, undocumented but the WebUI looks compliant with the N60's API)
 - N60: v2.02.0008 ([API v2.1:](https://newen.kiloview.com/docs/n60/api/))

Without testing, I would expect this for the other models, due to the API version changes:
 - N40/N40 Plus - Should work as it's [API v1.0](https://newen.kiloview.com/docs/n40/api/)
 - N5/N6 - Most likely also will not, as they are now on [API v3.0](https://newen.kiloview.com/docs/n5-n6-for-ndi/api/) on the newest firmware

**Comment by lenartd2** [comment]
Heads up for anyone finding this: the connection failure here is specifically an **N60** issue. I looked into this while fixing a related-but-different problem on N5/N6, and it's worth separating the two clearly since they don't share a fix:

- **N60 (this issue, and #14)**: runs Kiloview API **v2.1**. This module's client was written for the older v1.0 conventions (used by N40) and was never updated for v2.1's changes, which is why login/connection fails here. #30 correctly identifies this and proposes removing N60 from the supported products list until someone with N60 hardware can fix the v2.1 conventions specifically (different base URL/auth/endpoints than either v1.0 or v3.0).
- **N5/N6 on current firmware**: run a different API generation, **v3.0**, with its own separate set of breaking changes. That's what #36 fixes - it doesn't touch N60/v2.1 at all, so it won't help this specific report.

So if you're hitting this on an N60: #30 is the relevant one for now (it's honest that N60 isn't supported by the current client rather than silently failing), and N60/v2.1 support remains an open gap - happy to help if someone with that hardware wants to work through it the same way #36 does for N5/N6.

If you're hitting a similar-looking "device unreachable"/connection error on an **N5 or N6** instead, #36 is very likely your fix.
